pkg/config/types/v1alpha1/v1alpha1_types.go

```diff
@@ -343,6 +343,14 @@ type ClusterConfig struct {
 	//         - "https://www.mysweethttpserver.com/manifest2.yaml"
 	ExtraManifests []string `yaml:"extraManifests,omitempty"`
 	//   description: |
+	//     A list of key:value pairs that will be added while fetching the ExtraManifests.
```

Comment:
Would it make more sense to just have `Headers` be an optional part of a larger ExtraManifests struct? It seems likely that I would want headers for some, but not necessarily all, of the manifests I wish to deploy. I'm thinking something like this in the yaml:

```yaml
extraManifests:
  - url: http://test.com/file.yaml
    headers:
      - foo: bar
      - baz: test
  - url: http://test2.com/file2.yaml
```

Reply:
Yeah I like this. It does make things a bit more verbose if each URL has the same headers, but we don't lose the ability to have different headers if we need to.

Reply:
Are we ok with this as it would break v1alpha1 config?

Reply:
Oh, I guess I didn't realize that it broke `v1alpha1` 🤔 . In that case, then I suppose the current implementation would have to work.

Reply:
My only nit in that case, is that I think `ExtraManifestHeaders` should be a map. Then we don't have to split based on `:`.